docs/rfc/rfc-7.adoc

```diff
@@ -0,0 +1,237 @@
+
+=== Implementation
+
+==== Coordinator
+
+We introduce a rotating role of Coordinator responsible for proposing a sweep
+Bitcoin transaction. The role of Coordinator rotates every 8 hours across all
+operators in the group. For example, if the group consist of the following
+members: `[0xA, 0xB, 0xC, 0xC, 0xB, 0xD`] the rotation rounds are: `0xA, 0xB,
+0xC, 0xD`. The rotation schedule begins at UNIX epoch (00:00:00 UTC on 1 January
+1970).
+
+==== Sweeping schedule
+
+Sweep can only be proposed by the current Coordinator. The Coordinator can only
+propose a sweep between:
+
+- 00:00 UTC - 0:30 UTC
+- 08:00 UTC - 08:30 UTC
+- 16:00 UTC - 16:30 UTC
+
+If the current Coordinator does not propose a sweep within the given time
+window, the attempt is skipped.
+
+The Coordinator must not propose a sweep if there is currently a pending signing
+operation being executed by the wallet or if the last signed transaction from
+the wallet does not have 6 Bitcoin confirmations yet.
+
+The coordinator may slightly delay broadcasting the proposition to give all
+clients in the group a chance to properly recognize a Coordinator rotation.
+
+==== Sweep transaction
+
+The Coordinator chooses sufficiently confirmed deposits for the sweep. To avoid
+synchronization problems with deposits on the brink of being sufficiently
+confirmed, the Coordinator may choose deposits with more than 6 confirmations
+on Bitcoin and 20 confirmations on Ethereum. To avoid problems with proving
+large transactions and too many signatures to be executed, the Coordinator
+should not pick more than 10 deposits. The Coordinator chooses the Bitcoin
+transaction fee reflecting the current situation in the Bitcoin network. To
+leave enough time for retries, deposits for which Bitcoin transactions unlock
+for a refund earlier than two weeks from now are skipped.
+
+The Coordinator must not propose a sweep if there is currently a pending signing
+operation being executed by the wallet or if the last transaction from the
+wallet does not have 6 Bitcoin confirmations yet.
+
+The Coordinator proposes a sweep over the wallet's broadcast channel.
+
+All signing group members must confirm that all deposits in the proposed sweep
+are sufficiently confirmed, that the maximum number of deposits is not exceeded,
+that the proposed fee reflects the current situation in the Bitcoin network
+plus/minus some margin, that all deposit UTXOs can be unlocked with the
+wallet's public key, and that there is enough time to unlock UTXO before the
+refund.
+
+When proposing a sweep, the Coordinator sets the Ethereum block number at which
+the signing protocol should start. All clients validate if that block's number
+is +-2 blocks from the current one.
+
+If all these requirements are met, the last transaction executed by the wallet
+is sufficiently confirmed, and there is no other signing by the wallet in
+progress, the signing group members proceed with tECDSA signing protocol.
```

Comment:
Let's say the Coordinator proposed a sweep at block `x` and set the time of signing protocol start to `x+2`. Is there a possibility that some separate signing would start at `x+1`? For example a signing of redemption of some older deposits?

Reply:
Nope. Only one signing may be in progress and the operators announce their readiness to sign something.